### ISSUE TITLE
Use region from aws configuration by default

### ISSUE THREAD

**Issue by miguelgrinberg** [pynamodb/PynamoDB#211]
It would be nice if pynamodb used the region I have in my ~/.aws/credentials file by default. That's the behavior I get from boto and the aws cli. Instead, pynamodb tables that don't have a region explicitly stated always go to us-east-1.

**Comment by yedpodtrzitko** [comment]
I can imagine the advantage on the dev machine, but how would that work in a production? Would you need to deploy `~/.aws/credentials` as well? Sounds cumbersome.

**Comment by miguelgrinberg** [comment]
@yedpodtrzitko I'm not sure I understand why that would be cumbersome.

Boto has a procedure by which it loads credentials ([link](http://boto3.readthedocs.io/en/latest/guide/configuration.html#configuring-credentials)). It doesn't always have to be a ~/.aws/credentials file, there are many other ways to feed credentials to boto, and all support passing a region. By taking a hardcoded choice for us-east-1, the standard procedure is being altered.

In your example of a production machine, say an EC2 instance, I would want credentials to be deployed to the EC2 metadata, so that there is no need to mess with actual credential files. I haven't tested this, but what would happen if I deploy an EC2 instance with a role from another region attached, then try to use pynamodb from that EC2? The id and secret will be picked up from the instance metadata, but then pynamodb will try to access us-east-1, which may or may not work depending on what role is attached to the instance. What I think pynamodb should do, is let boto3 figure out a default region using its own mechanisms.

**Comment by brandond** [comment]
I just do something like this:

```python
from pynamodb.models import Model
from botocore.session import Session

class MyModel(Model):
    class Meta:
        table_name = 'my-model'
        region = Session().get_config_variable('region')
    myKey = UnicodeAttribute(hash_key=True)
```

That will use the region from the default profile, Lambda environment, etc.

**Comment by jordal** [comment]
I agree this would be a nice feature. Would anyone volunteer to turn @brandond's snippet into a PR that always sets the default region appropriately?

**Comment by jpinner-lyft** [comment]
@danielhochman @garrettheel could the same effect be achieved by simply removing pynamo's default of the region?

I would naively expect passing `None` to botocore's `create_client` call to cause it to go through its normal configuration steps, including using a region from the environment or from a credentials file if one is configured.

I don't believe botocore's lookup includes retrieving instance metadata as done in #330, but if it does not, I don't really think that is something we should include (a user can still choose to configure the region in the model's Meta class using that metadata).

I realize that removing `us-east-1` as the default is a breaking change but IMHO it is more in line with how most aws tools work (i.e. they require a region to be specified somewhere).

**Comment by sebastianseitz** [comment]
While I personally like the metadata approach, I see your point. `botocore` seems to read the env-variable if nothing else is set as `region_name` in the session (https://github.com/boto/botocore/blob/30c96a24796e423b25108e9c72d2c541d4bdf617/botocore/session.py#L83)

So this should be fine, I guess.

**Comment by jmphilli** [comment]
i agree with @jpinner-lyft 
we should remove the default region, update the docs with an example of how to set it and just use botocore flags/config

i do feel like this should be packaged in a 4.0 type version so that users don't upgrade unexpectedly and fail to communicate with the DB

**Comment by tedivm** [comment]
Bump! We ran into this issue today and it would be great if this could get resolved.

**Comment by spcogg** [comment]
Another bump, it's 2020 and it's still an issue for anyone that doesn't live in the us-east-1 region!

**Comment by kislyuk** [comment]
I just got bitten by this. It is not correct behavior for pynamodb to default to the us-east-1 region; it needs to accept the external configuration in the absence of any explicit user-supplied configuration. Please, can we get this fixed?

**Comment by CodePint** [comment]
Bump, also got bitten with this. Can we please get this implemented.

**Comment by jhmartin** [comment]
This was a surprise to me, I expected it to default to current-region.

**Comment by jalaziz** [comment]
We got bit by this too. I've put up a PR to remove the fixed default region (#1003).
